### ISSUE TITLE
Categorical Columns Handling + Few specific questions

### ISSUE THREAD

**Issue by Daniel8hen** [shenweichen/DeepCTR-Torch#99]
Hi,
I'd like to ask few questions about the algorithm:
1. Categorical Columns handling - 
In case I have a dataset of appID, advertisementID, country and deviceOS, and label (CTR). - e.g. 123, 1234, US, Android, 0.15 (some aggregation level).
Also, a categorical value can be "userId" for example (userId <> advertisementID - instead appId <> advertisementID)
How does the appID and advertisementID being handled?
I read that the embedding you guys are doing is good for high cardinality values (e.g. appId / advID - thousands of possible values).

2. Data Limitations:
Also, is there any limitation to number of categorical columns? Number of columns overall? Number of Data points? 
I'm asking this as I am used to use Spark up till now with distributed training, so I am wondering what could be the best way for me to migrate.

3. Unseen values:
Also, what about categories that were not in the train but are in the test? How should one tackle them?
I'd love to even go on a call with any of you just to make those questions much more clear, so maybe then I'll be able to add some documentation to your site about "this is the base dataset" ---> "this is the new dataset", or something like that.

**Comment by zanshuxun** [comment]
Thanks for your attention!
1. All categorical features (e.g. your userId, appID, advertisementID) will be transformed to numerical values by `sklearn.preprocessing.LabelEncoder` and then embeded.
2. This may depend on the memory of your workers. 
3. Sparse features (i.e. categorical features) are usually hashed, then we can make recommendations for unseen values. Unfortunately, Feature Hashing is not supported in torch version, you can use tensorflow version of [DeepCTR](https://github.com/shenweichen/DeepCTR). Here is a example: https://github.com/shenweichen/DeepCTR/blob/e9c8f08f86/examples/run_classification_criteo_hash.py#L25
